test/export/opinfo_schema.py

```diff
@@ -0,0 +1,109 @@
+# Owner(s): ["oncall: export"]
+
+import torch
+from torch._dispatch.python import enable_python_dispatcher
+from torch._subclasses.schema_check_mode import SchemaCheckMode
+from torch.fx.operator_schemas import normalize_function
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    ops,
+)
+from torch.testing._internal.common_methods_invocations import op_db
+from torch.testing._internal.common_utils import TestCase
+from torch.utils._pytree import tree_map
+
+# Simplified naming for C++ classes
+SchemaArgument = torch._C._SchemaArgument
+SchemaArgType = torch._C._SchemaArgType
+SchemaInfo = torch._C._SchemaInfo
+
+test_classes = {}
+
+
+class PreDispatchSchemaCheckMode(SchemaCheckMode):
```

Comment:
Hmm, is the implementation different enough to warrant a new class? Another option is just to use the existing `SchemaCheckMode`, and give it a new `bool is_predispatch` argument, that you use to optionally set the `_dispatch_key` field on it.

Reply:
I think the main reason is to have access to the aten op to check if it's aliasing or mutating before we raise an error? Without subclassing I'm not sure how to access the op and check from the OpInfo object.